### PR TITLE
Fix File.writeString failing with Flash/AIR

### PR DESCRIPTION
**Pull request by ghost**
Using touch causes crashes in File, this is because File.saveContent is not meant to be used in AIR/Flash for cross-platform filesystem operations. Tested in Flash and OpenFL. No issues found.

### CODE REVIEW

File: src/hx/files/File.hx [hunk 1]
```diff
@@ -479,7 +479,16 @@ class File {
          return;
 
       #if (sys || macro || nodejs)
+         #if (flash || openfl)
+         var dir:String = haxe.io.Path.directory(path.toString());
```
Comment (sebthom) [inline src/hx/files/File.hx:485]: Thanks for your contribution! could you add a short inline comment describing the reason for this extra code block?

Comment (ghost) [reply src/hx/files/File.hx:485]: Sure. Just a moment.

File: src/hx/files/File.hx [hunk 2]
```diff
@@ -479,7 +479,19 @@ class File {
...
       #if (sys || macro || nodejs)
+         #if (flash || openfl)
+         //before writing a string, make sure path's parent exists.
+         //if it doesn't, it needs to be created to write the content for OpenFL and AIR/Flash.
+         var dir:String = haxe.io.Path.directory(path.toString());
+         if (!FileSystem.exists(dir))
+            FileSystem.createDirectory(dir);
+         var output = sys.io.File.write(path.toString(), false);
+         output.writeString(content);
+			output.close();
```
Comment (sebthom) [inline src/hx/files/File.hx:490]: Could `sys.io.File.saveContent(path.toString(), content);` be use instead after the directory was created or does `sys.io.File.write` need to be used?

Comment (ghost) [reply src/hx/files/File.hx:490]: Tested `sys.io.File.saveContent(path.toString(), content)`, but it did not work. I had to use `File.write`.

Comment (sebthom) [reply src/hx/files/File.hx:490]: ok, thanks

**Pull request merged**